### DISCUSSION TITLE
closed: deadline exceeded while waiting connection

### DISCUSSION THREAD

**Discussion by hy7635** [bluenviron/mediamtx#3383] [Questions]
### Question

Hello, deploy mediamtx on the local Windows platform, and you can access the webrtc address to open the surveillance video stream on both the local and local networks;
Deploying the mediamtx service to a public cloud server will result in a closed: deadline exceeded while waiting connection error after 10 seconds. What is the reason for this error? How to solve it
The IP shown in the picture should be the company's public network exit IP address

2024/05/24 11:43:28 INF [WebRTC] [session 9650411a] created by 223.76.255.138:55935
2024/05/24 11:43:38 INF [WebRTC] [session 9650411a] closed: deadline exceeded while waiting connection
2024/05/24 11:43:46 INF [WebRTC] [session c67b1950] created by 223.76.255.138:55935
2024/05/24 11:43:56 INF [WebRTC] [session c67b1950] closed: deadline exceeded while waiting connection

![1716521945786](https://github.com/bluenviron/mediamtx/assets/43111806/9990edaf-60b7-4b1e-9e57-0ba3100c8f23)

**Comment by U29** [comment]
I had same problem.

check here to solve:
https://github.com/bluenviron/mediamtx/discussions/3220#discussioncomment-9551437

**Comment by aler9** [comment]
Instructions on how to restore WebRTC connectivity when exposing the server publicly are available in the README:
https://github.com/bluenviron/mediamtx?tab=readme-ov-file#solving-webrtc-connectivity-issues

[a comment by an automated account is not shown]
